apps/election-manager/src/components/ExportElectionBallotPackageModalButton.tsx

```diff
@@ -104,23 +108,32 @@ const ExportElectionBallotPackageModalButton: React.FC = () => {
   )
 
   // Callback to open the file dialog.
-  const openFileDialog = async () => {
+  const saveFileCallback = async (openDialog: boolean) => {
     if (state.type !== 'ArchiveBegin') {
       throw new Error(
-        `unexpected state '${state.type}' found during openFileDialog callback`
+        `unexpected state '${state.type}' found during saveFileCallback`
       )
     }
     try {
-      await state.archive.begin({
-        defaultPath: defaultFileName,
-        filters: [{ name: 'Archive Files', extensions: ['zip'] }],
-      })
+      const usbPath = await getDevicePath()
+      const pathToFolder = path.join(usbPath, EXPORT_FOLDER)
+      const pathToFile = path.join(usbPath, EXPORT_FOLDER, defaultFileName)
+      if (openDialog) {
+        await state.archive.beginWithDialog({
+          defaultPath: pathToFile,
+          filters: [{ name: 'Archive Files', extensions: ['zip'] }],
+        })
+      } else {
+        await window.kiosk!.makeDirectory(pathToFolder, {
+          recursive: true,
+        })
```

Comment:
Should this be in the `beginWithDirectSave` call? This code doesn't otherwise deal with `window.kiosk`, right?